### PR TITLE
BIGTOP-3356: Add CentOS 8 support to the Docker provisioner

### PR DESCRIPTION
**Pull request by sekikn**

### CODE REVIEW

**Comment by iwasakims** [comment]
@sekikn I got the following error on provisioning with the patch. `yum-config-manager` seems to be replaced by `yum config-manager` on CentOS 8.
```
sudo: yum-config-manager: command not found
```

**Comment by iwasakims** [comment]
After I fixed the yum-config-manager part, I got the another error. Installing `rng-tools` seemed to be failed.
```
Failed to download metadata for repo 'AppStream'
Error: Failed to download metadata for repo 'AppStream'
sed: can't read /usr/lib/systemd/system/rngd.service: No such file or directory
Failed to start rngd.service: Unit rngd.service not found.
```

**Comment by iwasakims** [comment]
@sekikn Sorry, the cause of the error was that I did not update docker image by build.sh.

File: provisioner/docker/config_centos-8.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+docker:
+        memory_limit: "4g"
+        image: "bigtop/puppet:trunk-centos-8"
+
+repo: "http://repos.bigtop.apache.org/releases/1.5.0/centos/8/$basearch"
```
Comment (iwasakims) [inline provisioner/docker/config_centos-8.yaml:20]: Is this URL invalid before the release of 1.5.0? I got the the following error.
```
Error: Execution of '/usr/bin/dnf -d 0 -e 1 -y install java-1.8.0-openjdk-devel' returned 1: Error: Failed to download metadata for repo 'Bigtop_0'
Error: /Stage[main]/Jdk/Package[jdk]/ensure: change from 'purged' to 'present' failed: Execution of '/usr/bin/dnf -d 0 -e 1 -y install java-1.8.0-openjdk-devel' returned 1: Error: Failed to download metadata for repo 'Bigtop_0'
```

**Comment by iwasakims** [comment]
Hmm... starting services by `systemctl` does not work in docker container. Since it works outside container, I'm suspecting issue like https://github.com/docker/for-linux/issues/835.

**Comment by iwasakims** [comment]
adding /sys/fs/cgroup to volumes worked for me to fix the `systemctl` issue.
```
diff --git a/provisioner/docker/docker-compose.yml b/provisioner/docker/docker-compose.yml
index cee849e9..b3bb1a42 100644
--- a/provisioner/docker/docker-compose.yml
+++ b/provisioner/docker/docker-compose.yml
@@ -24,3 +24,4 @@ bigtop:
     - ./config/hiera.yaml:/etc/puppet/hiera.yaml
     - ./config/hieradata:/etc/puppet/hieradata
     - ./config/hosts:/etc/hosts
+    - /sys/fs/cgroup:/sys/fs/cgroup:ro
```

File: provisioner/docker/config_centos-8.yaml (hunk 1 above)
Comment (sekikn) [reply provisioner/docker/config_centos-8.yaml:20]: That's right, just the same as [config_fedora-31.yaml](/apache/bigtop/blob/master/provisioner/docker/config_fedora-31.yaml). I replaced that URL with the existing one temporally when I tested it.

**Comment by sekikn** [comment]
Great, will try it locally. Thanks!

**Comment by evans-ye** [comment]
> adding /sys/fs/cgroup to volumes worked for me to fix the `systemctl` issue.
> 
> ```
> diff --git a/provisioner/docker/docker-compose.yml b/provisioner/docker/docker-compose.yml
> index cee849e9..b3bb1a42 100644
> --- a/provisioner/docker/docker-compose.yml
> +++ b/provisioner/docker/docker-compose.yml
> @@ -24,3 +24,4 @@ bigtop:
>      - ./config/hiera.yaml:/etc/puppet/hiera.yaml
>      - ./config/hieradata:/etc/puppet/hieradata
>      - ./config/hosts:/etc/hosts
> +    - /sys/fs/cgroup:/sys/fs/cgroup:ro
> ```

@iwasakims Can you share some technical reference to this. Thanks!

**Comment by iwasakims** [comment]
> Can you share some technical reference to this. 

@evans-ye The background is similar to [BIGTOP-3302](https://issues.apache.org/jira/browse/BIGTOP-3302). systemd added check to see if service process belongs to the service as a fix for [CVE-2018-16888](https://nvd.nist.gov/vuln/detail/CVE-2018-16888). The check uses cgroups.
* https://github.com/systemd/systemd/issues/6632
* https://github.com/systemd/systemd/pull/7816

If cgroups info is not available, init script fails with following message, indicating failure of checking added by the https://github.com/systemd/systemd/issues/6632.
```
May 21 11:53:07 2eef6a902086 zookeeper-server[2620]: Starting zookeeper ... STARTED
May 21 11:53:07 2eef6a902086 runuser[2665]: pam_unix(runuser:session): session closed for user zookeeper
May 21 11:53:07 2eef6a902086 systemd[1]: zookeeper-server.service: New main PID 2714 does not belong to service, and PID file is not owned by root. Refusing.
```

**Comment by sekikn** [comment]
Thanks @iwasakims, I also confirmed your fix worked well. I updated the PR.

**Comment by sekikn** [comment]
Thanks! Merged into master.

**Comment by evans-ye** [comment]
The investigation is informative. Thanks @iwasakims.

**Pull request merged**